File: src/IotBbq.App/IotBbq.App/Services/IAlarmService.cs

namespace IotBbq.App.Services
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using System.Threading.Tasks;

    public enum AlarmPriority
    {
        Normal,
        High
    }

    public interface IAlarmService
    {
        event EventHandler<bool> AlarmStateChanged;

        bool IsAlarming { get; set; }

        void Silence();

        void TriggerAlarm(AlarmPriority priority, TimeSpan? duration = null);
    }
}

File: src/IotBbq.App/IotBbq.JsApp/src/app/services/IAlarmService.ts
import { InjectionToken } from "@angular/core";

export enum AlarmPriority {
  Normal,
  High
}

export interface IAlarmService {
  alarmStateChanged: (state: boolean) => void;
  isAlarming(): boolean;
  silence(): void;
  triggerAlarm(priority: AlarmPriority, duration?: number);
}

export const ALARM_SVC_TOKEN = new InjectionToken('ALARM_SVC_TOKEN');